pkg/resource/endpoint/hooks.go

```diff
@@ -0,0 +1,192 @@
+		return newValidationError("spec.eventBuses", "must contain exactly two event buses")
+	}
+
+	// event bus names must be identical
+	arns := make([]string, 2)
+	for i, b := range spec.EventBuses {
+		if b.EventBusARN == nil {
+			return newValidationError("spec.eventBuses", "event bus arn must be set")
+		}
+		arnInfo, err := arn.Parse(*b.EventBusARN)
+		if err != nil {
+			return newValidationError("spec.eventBuses", fmt.Sprintf("invalid arn %q", *b.EventBusARN))
+		}
+		arns[i] = arnInfo.Resource
+	}
+
+	if arns[0] != arns[1] {
+		return newValidationError("spec.eventBuses", "event bus names must be identical")
+	}
+	return nil
+}
+
+// endpointAvailable returns true if the supplied Endpoint is in an available
+// status
+func endpointAvailable(r *resource) bool {
+	if r.ko.Status.State == nil {
+		return false
+	}
+	state := *r.ko.Status.State
+	return state == svcsdk.EndpointStateActive
+}
+
+// endpointInMutatingState returns true if the supplied Endpoint is in the process of
+// being created
+func endpointInMutatingState(r *resource) bool {
+	if r.ko.Status.State == nil {
+		return false
+	}
+	state := *r.ko.Status.State
+	return state == svcsdk.EndpointStateCreating || state == svcsdk.EndpointStateUpdating || state == svcsdk.EndpointStateDeleting
+}
+
+// requeueWaitUntilCanModify returns a `ackrequeue.RequeueNeededAfter` struct
+// explaining the Endpoint cannot be modified until it reaches an available
+// status.
+func requeueWaitUntilCanModify(r *resource) *ackrequeue.RequeueNeededAfter {
+	if r.ko.Status.State == nil {
+		return nil
+	}
+	msg := fmt.Sprintf("Endpoint is in status %q, cannot be modified.", *r.ko.Status.State)
+	return ackrequeue.NeededAfter(
+		errors.New(msg),
+		defaultRequeueDelay,
+	)
+}
+
+// if an optional desired field value is nil explicitly unset it in the request
+// input
+func unsetRemovedSpecFields(
+	delta *ackcompare.Delta,
+	spec v1alpha1.EndpointSpec,
+	input *eventbridge.UpdateEndpointInput,
+) {
+	if delta.DifferentAt("Spec.Description") {
+		if spec.Description == nil {
+			input.SetDescription("")
+		}
+	}
+
+	if delta.DifferentAt("Spec.ReplicationConfig") {
+		if spec.ReplicationConfig == nil {
+			input.SetReplicationConfig(&eventbridge.ReplicationConfig{State: aws.String("ENABLED")})
+		}
+	}
+}
+
+func customPreCompare(
+	delta *ackcompare.Delta,
+	a *resource,
+	b *resource,
+) {
+	aDescr := a.ko.Spec.Description
+	bDescr := b.ko.Spec.Description
+
+	if !tags.EqualStrings(aDescr, bDescr) {
+		delta.Add("Spec.Description", aDescr, bDescr)
+	}
+
+	aRole := a.ko.Spec.RoleARN
+	bRole := b.ko.Spec.RoleARN
+
+	if !tags.EqualStrings(aRole, bRole) {
+		delta.Add("Spec.RoleARN", aRole, bRole)
+	}
+
+	aReplCfg := a.ko.Spec.ReplicationConfig
+	bReplCfg := b.ko.Spec.ReplicationConfig
+
+	if !equalReplicationConfigs(aReplCfg, bReplCfg) {
+		delta.Add("Spec.ReplicationConfig", aReplCfg, bReplCfg)
+	}
+}
+
+func equalReplicationConfigs(a, b *v1alpha1.ReplicationConfig) bool {
```

Comment:
You'll need to add a custom comparison for `Spec.EventBuses` too... (see my note above in `delta.go`)